File: command_mapb.go
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"log"
	"net/http"
)

func commandMapb(config *Config) {

	url := config.Previous

	if url == "" {
		url = "https://pokeapi.co/api/v2/location-area/"
	}
	res, err := http.Get(url)

	if err != nil {
		log.Fatal(err)
	}

	body, err := io.ReadAll(res.Body)
	defer res.Body.Close()

	if err != nil {
		log.Fatal(err)
	}

	var locations LocationAreaBatch
	err = json.Unmarshal(body, &locations)

	if err != nil {
		log.Fatal(err)
	}

	config.Next = locations.Next
	if locations.Previous != nil {
		config.Previous = *locations.Previous
	} else {
		config.Previous = ""
	}

	for i := range len(locations.Results) {
		fmt.Println(locations.Results[i].Name)
	}

}

File: src/commands/command_mapb.ts
import { stat } from "fs";
import { State } from "../state.js";

export async function commandMapb(state: State): Promise<void> {

    if (state.prevLocationURL === null) {
        console.log("you're on the first page");
        return;
    }

    const locations = await state.pokeAPI.fetchLocations(state.prevLocationURL);

    state.nextLocationURL = locations.next;
    state.prevLocationURL = locations.previous;

    for (const location of locations.results) {
        console.log(location.name);
    }
}